apps/studio/src/pages/dashboard/edit/index.tsx

```diff
@@ -69,18 +70,35 @@ const placeholder = {
 
 const EditPage: NextPageWithLayout = () => {
   const [isEditorOpen, setIsEditorOpen] = useState(true)
-  const [editorValue, setEditorValue] = useState(
-    JSON.stringify(placeholder, null, 2),
-  )
-  const [newEditorValue, setNewEditorValue] = useState({})
   const [editedSchema, setEditedSchema] = useState<any>(placeholder)
   const [isJSONValid, setIsJSONValid] = useState(true)
   const [isNewEditor, setIsNewEditor] = useState(false)
-  const [isCopied, setIsCopied] = useState(false)
 
+  const [isCopied, setIsCopied] = useState(false)
   const [jsonSchema, setJsonSchema] = useState<any>(null)
   const [validate, setValidate] = useState<any>(null)
 
+  const {
+    drawerState,
+    setDrawerState,
+    pageState,
+    setPageState,
+    editorState,
+    setEditorState,
+  } = useEditorDrawerContext()
+
+  const [{ theme, isGovernment, sitemap, name }] =
```

Comment:
suggestion(non-blocking): maybe consider just passing the props as a single object rather than destructuring here? Otherwise, we will have to make code-level changes if the schema changes.

I think there are some concerns with prop spreading so happy to discuss.

Reply:
swapped to props pread